test_util/helpers.py

```diff
@@ -37,6 +37,81 @@ def authenticate(self, cluster):
         logging.info('Authentication successful')
 
 
+class ApiClient:
+
+    def __init__(self, cluster, user, api_base, default_headers=None, ca_cert_path=None):
+        self.cluster = cluster
+        self.user = user
+        self.api_base = api_base
+        if default_headers is None:
+            default_headers = dict()
+        self.headers = default_headers
+        self.ca_cert_path = ca_cert_path
+        self.port_mapping = {
+            'master': {'http': 80, 'https': 443},
+            'agent': {'http': 61001, 'https': 61002},
+            'public_agent': {'http': 61001, 'https': 61002}}
+
+    def api_request(self, method, path, node=None, port=None, **kwargs):
+        """
+        Makes a request with default headers + user auth headers if necessary
+        If DCOS_CA_CERT_PATH is set in environment, this method will pass it to requests
+        Args:
+            method: (str) name of method for requests.request
+            node: see get_url()
+            path: see get_url()
+            port: see get_url()
+            **kwargs: any optional arguments to be passed to requests.request
+        """
+        headers = copy.copy(self.headers)
+        if self.cluster.auth_enabled and self.cluster.web_auth_default_user:
+            headers.update(self.cluster.web_auth_default_user.auth_header)
+
+        if self.ca_cert_path:
+            kwargs['verify'] = self.ca_cert_path
+
+        headers.update(kwargs.pop('headers', {}))
+        url = self.get_url(node=node, path=path, port=port)
+        logging.info('{}: {}'.format(method, url))
+        logging.debug('kwargs: {}'.format(kwargs))
+        logging.debug('headers: {}'.format(headers))
+        return requests.request(method, url, headers=headers, **kwargs)
+
+    def get_url(self, node, path, port=None):
+        """
+        Args:
+            path: (str) URL path to request if self.api_base is set it will be prepended
+            node: (str) the hostname of the node to be requested from, if node=None,
+                then public cluster address (see environment DCOS_DNS_ADDRESS)
+            port: (int) port to be requested at. If port=None, the default port
+                for that given node type will be used
+        Returns:
+            fully-qualified URL string for this API
+        """
+        if node is None:
+            node = self.cluster.dns_host
+            role = 'master'
+        else:
+            role = self.cluster.get_node_role(node)
+
+        if self.api_base:
+            path = '{}/{}'.format(self.api_base.rstrip('/'), path.lstrip('/'))
+
+        if port is None:
+            port = self.port_mapping[role][self.cluster.scheme]
+
+        return '{}://{}/{}'.format(self.cluster.scheme, '{}:{}'.format(node, port), path.lstrip('/'))
```

Comment:
very tiny nit: can we make this leave of the port if it's port 80 and http or port 443 and https? (Practically leaving it doesn't really harm and is actually more explicit, just from web browsers and the like I'm used to it being folded away).